### DISCUSSION TITLE
Adding space or a border around outer edge of a montage

### DISCUSSION THREAD

**Discussion by excedrin70** [ImageMagick/ImageMagick#7394] [Help] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

macOS 14.5

### Description

I have a folder with a couple hundred images.  They are all the same dimensions.  I am trying to create a montage with all of the images in a randomly rotated order.  This command seems to work almost perfectly except the images on the left hand side have some of their border cropped off.

Is there a way to add extra padding or space around the outer edge of the final montage so that the white border I am adding isn't cropped?  I have tried to use null, but it only seems to add a single "blank" image at the top.  It doesn't have to be a large amount of space, just enough so that none of the final polaroid images have their border missing.

magick montage *.jpg -tile 10x -thumbnail 400x400 -bordercolor white +polaroid -gravity center -background gray -geometry -5-5 polaroid_overlap2.jpg

**Comment by fmw42** [comment]
Use `-background white -gravity west -splice Xx0` where X is the number of columns you want to add on the left to an image. See https://imagemagick.org/Usage/crop/#splice

Have you seen https://imagemagick.org/script/formats.php#pseudo ASHLAR

**Reply by excedrin70** [reply]
Thank you for the reply.  Is that available in all versions of ImageMagick?  I am trying to use it and receive the following error.

unrecognized option `-splice' @ error/montage.c/MontageImageCommand/1597

I am using version 7.1.1-33

**Reply by fmw42** [reply]
Yes, it is available in all modern versions.  Please show your full command so we can inspect for typos.

**Reply by excedrin70** [reply]
I am using the command from my original post.  I have tried to place the splice option in different places with the same result.

magick montage *.jpg -tile 10x -thumbnail 400x400 -bordercolor white +polaroid -background gray -gravity west -splice 1x0 -geometry -5-5 polaroid_overlap2.jpg

**Reply by fmw42** [reply, marked as answer]
montage does not permit -splice as an option. You have to do that after the montage by piping the output of magick montage to the input of magick.

`magick montage inputs ..... miff:- | magick - -background gray -gravity west -splice 1x0 polaroid_overlap2.jpg
`

**Reply by excedrin70** [reply]
Thank you so much.  I really appreciate it!
